AK/StringImpl.cpp

```diff
@@ -172,4 +172,18 @@ void StringImpl::compute_hash() const
     m_has_hash = true;
 }
 
+void StringImpl::reverse()
```

Comment:
This is bad business. If you look at the rest of `StringImpl`, you'll notice that there's no way to mutate the contents of an existing string.
`StringImpl` is reference-counted, and so many `String` objects can share the same underlying `StringImpl`.
That means that if you called `reverse()` on one `String`, any number of other `String`s may now have become reversed as well, which is probably not what you wanted.

The appropriate API would be something that creates a new `StringImpl`:
`NonnullRefPtr<StringImpl> StringImpl::reversed() const`

Reply:
> That means that if you called `reverse()` on one `String`, any number of other `String`s may now have become reversed as well, which is probably not what you wanted.

Completely neglected to realise that. I've made the changes.